### ISSUE TITLE
Ability to distribute optional extras

### ISSUE THREAD

**Issue by zalmane** [Nuitka/Nuitka#1839]
I am creating an executable that will optionally use database drivers (using SQLAlchemy). I would like to create optional extra packages that can be installed on demand.
Is there a way to compile these drivers as separate packages and add them to the dist dir on demand? I was not clear if the extension module mechanism is meant for that or there is another way to provide them.
For example, consider the following:
```
from sqlalchemy import create_engine
import sqlalchemy.sql.default_comparator
db_string = "postgresql://postgres:postgres@localhost:5432/postgres"

db = create_engine(db_string)

# Read
result_set = db.execute("SELECT * FROM emp")
for r in result_set:
    print(r)
```

Ability to compile the base program without psycopg2 and optionally add it post-install as a separate .so or package.

**Comment by kayhayen** [comment]
You can include packages that are used dynamically, but the result will be static.

I assume, what would be nice, if wheels could be installed into a dist folder, I accept it as a feature request, but I do not see it happening unless somebody qualified hacks pip to do it for us.

I am keeping this open and add the help wanted tag, so people find this. But for now, the way to go is to use `--include-package` on all drivers you mean to support.

**Comment by zalmane** [comment]
can you explain what you mean by 'result will be static'?
can I remove the compiled drivers from the dist folder when shipping my
binary and optionally add them later? would that work or break the
installation?
I am mostly looking to reduce the install size of the basic package once
there are many drivers and dependencies in there.

**Comment by kayhayen** [comment]
No, you cannot at this time. Maybe somebody will add support for it.

**Comment by zalmane** [comment]
Am willing to look into contributing a PR. Can you give me a pointer into where to start looking?

**Comment by kayhayen** [comment]
https://github.com/Nuitka/Nuitka/blob/develop/CONTRIBUTING.md

**Comment by kayhayen** [comment]
@zalmane what is your status with this, there is nothing visible here

**Comment by zalmane** [comment]
hello,
we opted not to use Nuitka since we couldn't find a solution for optional
extras.
